### PR TITLE
Fixed typos in the codebase

### LINKED ISSUE #24466: Found Typos Using Codespell Library
I ran `codespell` over the SymPy codebase and found these typos. The issue could be labelled as a `good first issue`.
I have gone through the typos personally and checked for false positives. The file name and lines are shown below -
```

.\doc\src\explanation\glossary.md:701: falsy ==> false
.\doc\src\guides\solving\solve-ode.md:230: explicity ==> explicitly
.\sympy\combinatorics\perm_groups.py:3303: Procede ==> Proceed
.\sympy\physics\mechanics\jointsmethod.py:267: solveable ==> solvable
.\sympy\physics\mechanics\tests\test_kane.py:104: consisten ==> consistent
.\sympy\polys\matrices\domainmatrix.py:906: substract ==> subtract
.\sympy\polys\matrices\domainmatrix.py:912: Substraction ==> Subtraction
.\sympy\utilities\mathml\data\mmlctop.xsl:263: sematics ==> semantics
.\sympy\utilities\mathml\data\simple_mmlctop.xsl:264: sematics ==> semantics
```

### PR DESCRIPTION
**Pull request by jonathanyanez02**
# Typos
* Fixed the 8 typos found in separate files

### CODE REVIEW

**Comment by 1e9abhi1e10** [comment]
@jonathanyanez02 , the failure in the `test authors` is obvious you should add your name and email in the `.mailmap` file according to the following [instructions](https://github.com/sympy/sympy/wiki/Development-workflow#add-your-name-and-email-address-to-the-mailmap-file).
Also, Improve the release note that tells to anyone meaningfully what this is about. Read [here](https://github.com/sympy/sympy/wiki/Writing-Release-Notes) for writing the release notes.

**Comment by jonathanyanez02** [comment]
Issue is resolved

**Comment by faze-geek** [comment]
Hi @jonathanyanez02 
An appropriate heading for the pull request would be something like `"Fixed Typos In the Codebase"` instead of copying the issue title itself.
I see you have not followed the OP format. Kindly checkout some completed pull requests to get an idea. The issue you have solved must be referred and you could add `NO ENTRY` as far as the release notes are concerned.

File: doc/src/explanation/glossary.md [hunk 1]
```diff
@@ -698,7 +698,7 @@ Three-valued logic
     but `None OR True` is `True` because the predicate is `True` whether the
     `None` "really" represents a value of `True` or `False`. One must be
     careful when using the usual Python logical operators like `and`, `or` and
-    `not` on three-valued logic, since `None` is falsy. See [the guide for
+    `not` on three-valued logic, since `None` is false. See [the guide for
```
Comment (asmeurer) [inline doc/src/explanation/glossary.md:701]: This change is incorrect.

File: doc/src/guides/solving/solve-ode.md [hunk 2]
```diff
@@ -227,7 +227,7 @@ cannot be explicitly expressed as $f(x)$ equaling a function of $x$. Instead,
 solution](https://en.wikipedia.org/wiki/Implicit_function).
 
 When `dsolve` returns an implicit solution, extracting the right-hand side of
-the returned equality will not give an explicity expression for the function to
+the returned equality will not give an explicitly expression for the function to
```
Comment (asmeurer) [inline doc/src/guides/solving/solve-ode.md:230]: This change is incorrect (should be "explicit")

**Comment by asmeurer** [comment]
Changes like this are fine, but please actually read the surrounding text and try to see if the change is correct or not, rather than blindly following the tool, especially for something like a spell checker which tends to have a lot of false positives.

File: sympy/utilities/mathml/data/mmlctop.xsl [hunk 3]
```diff
@@ -260,7 +260,7 @@ CONSTANT and SYMBOL ELEMENTS
     </xsl:when>
     <xsl:when test="($SEM_SW=$SEM_XREF or $SEM_SW=$SEM_XREF_EXT) and @id">
       <xsl:choose>
-        <xsl:when test="self::sematics">
+        <xsl:when test="self::semantics">
```
[the remark this answers is not in the document]
Comment (jonathanyanez02) [reply sympy/utilities/mathml/data/mmlctop.xsl:263]: Aside from the "explicit" change, I was just following what the issue was asking for. Completely new to this so sorry in advance

File: doc/src/guides/solving/solve-ode.md (hunk 2 above)
Comment (faze-geek) [reply doc/src/guides/solving/solve-ode.md:230]: Yes this correction was mentioned in the issue too . @jonathanyanez02 kindly correct it .

**Comment by jonathanyanez02** [comment]
Still learning. Thank you for your feedback. This is well appreciated

**Comment by sylee957** [comment]
I would close this because https://github.com/sympy/sympy/pull/24469 had passed the test first

**Pull request closed without merging**